site/app/entities/poll/Poll.php

```diff
@@ -91,39 +103,52 @@ public function getQuestion(): string {
     public function setQuestion(string $question): void {
         $this->question = $question;
     }
-
-    public function getStatus(): string {
-        return $this->status;
+    public function setClosed(): void {
+        $this->end_date = new \DateTime(DateUtils::BEGINING_OF_TIME);
     }
-
     public function setOpen(): void {
-        $this->status = "open";
+        $this->end_date = new \DateTime(DateUtils::MAX_TIME);
+    }
+    public function setEnded(): void {
+        $temp = DateUtils::getDateTimeNow();
+        $tempString = $temp->format('Y-m-d');
+        $this->end_date = new DateTime($tempString);
     }
-
     public function isOpen(): bool {
-        return $this->status === "open";
+        $now = DateUtils::getDateTimeNow();
+        return $now < $this->end_date;
     }
 
-    public function setClosed(): void {
-        $this->status = "closed";
+    public function isEnded(): bool {
+        $now = DateUtils::getDateTimeNow();
+        $closeDate = '1900-02-01T00:00:00';
+        return $now > $this->end_date && $this->end_date->format('Y-m-d\TH:i:s') !== $closeDate;
     }
 
     public function isClosed(): bool {
-        return $this->status === "closed";
+        $now = DateUtils::getDateTimeNow();
+        return $now > $this->end_date && $this->end_date->format('Y-m-d') === '1900-02-01';
```

Comment:
Same comment here.  It would be better to compare against a DateTime object.

Reply:
Fixed

Reply:
Reverted, but same as last comment.